support-frontend/assets/helpers/abTests/abtest.ts

```diff
@@ -219,7 +219,8 @@ function getParticipations(
 		 * if the  the ab test definition has includeOnlyContributionsOnlyCountries as true
 		 */
 		if (
-			selectedAmountsVariant?.testName !== contributionsOnlyAmountsTestName &&
+			selectedAmountsVariant &&
```

Comment:
`selectedAmountsVariant` is still an optional parameter within this call. This ensures one exists for the not equal check below👇

Reply:
What if `selectedAmountsVariant` is undefined and  `includeOnlyContributionsOnlyCountries` is `true` for the test? It looks like we'd now include a user in the `includeOnlyContributionsOnlyCountries` AB test in this case? Not sure that's what we'd want to do. 

Before this change we would have not allocated users to a test where `includeOnlyContributionsOnlyCountries` is `true`, if:

- `selectedAmountsVariant` was undefined OR
-  `selectedAmountsVariant.testName` was not `contributionsOnlyAmountsTestName`

Which would be what I'd expect? We can probably right a unit test to confirm this?

Reply:
Yes thats a case I had not thought of. I have added the test change and the addition to `abtest.ts` to pick this up this scenario. Hopefully looks better now?

Reply:
OK good to see and AB test to cover this scenario, but I actually don't think you need the extra if condition you've added below to handle it:

```
// Exclude undefined users assigned to the contributions only amounts test
		if (!selectedAmountsVariant && includeOnlyContributionsOnlyCountries) {
			return;
		}
```

What's the issue with the original if check without the `selectedAmountsVariant &&` condition? I think this original condition was enough:

```
		if (
			selectedAmountsVariant?.testName !== contributionsOnlyAmountsTestName &&
			includeOnlyContributionsOnlyCountries
		) {
			return;
		}
```

Reply:
updated and removed unecessary condition